packages/e2e-tests/src/specs/editor/fontCheck/fontCheckMetrics.js

```diff
@@ -0,0 +1,205 @@
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  createNewStory,
+  insertStoryTitle,
+  publishStory,
+  addRequestInterception,
+} from '@web-stories-wp/e2e-test-utils';
+
+async function createStoryWithTitle(title) {
+  await createNewStory();
+  await insertStoryTitle(title);
+  await expect(page).toClick('#library-tab-text');
+  const insertButton = await page.waitForXPath(
+    `//button//span[contains(text(), 'Title 1')]`
+  );
+  await insertButton.click();
+  await expect(page).toMatchElement('[data-testid="textFrame"]', {
+    text: 'Title 1',
+  });
+}
+
+async function updateFont(fontFamily) {
+  // Open style pane
+  await expect(page).toClick('li[role="tab"]', { text: /^Style$/ });
+  await expect(page).toClick('button[aria-label="Font family"]');
+  await page.keyboard.type(fontFamily);
+  await page.keyboard.press('ArrowDown');
+  await page.keyboard.press('Enter');
+}
+
+async function addStoryWithFont(title) {
+  await createStoryWithTitle(title);
+  await updateFont('Rock Salt');
+  await publishStory();
+  // Somehow reloading the page after creating the story here helps
+  // with making the test more stable, i.e. toggling dev tools
+  // works more reliably.
+  await page.reload();
+}
+
+function isPlatformMacOS() {
+  return page.evaluate(() => {
+    const { platform } = window.navigator;
+    return platform.includes('Mac') || ['iPad', 'iPhone'].includes(platform);
+  });
+}
+
+async function toggleDevTools() {
+  const areDevToolsOpen = Boolean(await page.$('#web-stories-editor textarea'));
+
+  // Cancel whatever current action to ensure the below shortcut works.
+  await page.keyboard.press('Escape');
+  await page.click('[aria-label="Web Stories Editor"]');
+
+  const isMacOS = await isPlatformMacOS();
+
+  await page.keyboard.down(isMacOS ? 'Meta' : 'Control');
+  await page.keyboard.down('Shift');
+  await page.keyboard.down('Alt');
+  await page.keyboard.down('J');
+  await page.keyboard.up('J');
+  await page.keyboard.up('Alt');
+  await page.keyboard.up('Shift');
+  await page.keyboard.down(isMacOS ? 'Meta' : 'Control');
+
+  // Dev Tools were not open before, but now they should be open.
+  if (!areDevToolsOpen) {
+    await page.waitForSelector('#web-stories-editor textarea');
+  }
+}
+
+async function getCurrentStoryData() {
+  await toggleDevTools();
+
+  const textareaContent = await page.evaluate(
+    () => document.querySelector('#web-stories-editor textarea').value
+  );
+
+  await toggleDevTools();
+
+  return JSON.parse(textareaContent);
+}
+
+describe('Font Check Metrics', () => {
+  let stopRequestInterception;
+  let mockResponse;
+
+  beforeAll(async () => {
+    await page.setRequestInterception(true);
+    stopRequestInterception = addRequestInterception((request) => {
+      if (request.url().includes('web-stories/v1/fonts') && mockResponse) {
+        request.respond(mockResponse);
+        return;
+      }
+
+      request.continue();
+    });
```

Comment:
Still of mocking the response like this, I think it might be better to do this in PR with a little e2e plugin.

Reply:
I actually find this to be a pretty good use case for request interception, since it's only needed for a part of the test.

A plugin is more useful if it's needed for the whole duration of the test, which is not the case here.

One drawback is that request interception does currently not work in Firefox, but I don't see us running tests against Firefox anytime soon.